doc/configuration.md

```diff
@@ -1,4 +1,289 @@
+
+- ```error```
+- ```warn```
+- ```info```
+- ```debug```
+
+## storage
+
+```yaml
+storage:
+	filesystem:
+		rootdirectory: /tmp/registry
+	azure:
+		accountname: <string>
+		accountkey: <string>
+		container: <string>
+	s3:
+		accesskey: <string>
+		secretkey: <string>
+		region: <string> # region name
+		bucket: <string>
+		encrypt: <bool>
+		secure: <bool>
+		v4auth: <bool>
+		chunksize: <int64>
+		rootdirectory: <string> 
+```
+
+The storage option is **required** and defines which storage backend is in use. At the moment only one backend may be configured, an error is returned when the registry is started with more than one storage backend configured.
+
+The following backends may be configured, **all options for a given storage backend are required**:
+
+### filesystem
+
+This storage backend uses the local disk to store registry files. It is ideal for development and may be appropriate for some small scale production applications.
+
+- rootdirectory: **Required** - This is the absolute path to directory in which the repository will store data.
+
+### azure
+
+This storage backend uses Microsoft's Azure Storage platform. 
+
+- accountname: **Required** -
+- accountkey: **Required** -
+- container: **Required** -
+
+### S3
+
+This storage backend uses Amazon's Simple Storage Service (a.k.a. S3).
+
+- accesskey: **Required** - Your AWS Access Key
+- secretkey: **Required** - Your AWS Secret Key.
+- region: **Required** - The AWS region in which your bucket exists. For the moment, the Go AWS library in use does not use the newer DNS based bucket routing.
+- bucket: **Required** - The bucket name in which you want to store the registry's data.
+- encrypt:
+- secure:
+- v4auth: This indicates whether Version 4 of AWS's authentication should be used. Generally you will want to set this to true.
+- chunksize: 
+- rootdirectory: **Optional** - This is a prefix that will be applied to all S3 keys to allow you to segment data in your bucket if necessary.
+
+## auth
+
+```yaml
+auth:
+	silly:
+		realm: silly-realm
+		service: silly-service
+	token:
+		realm: token-realm
+		service: token-service
+		issuer: registry-token-issuer
+		rootcertbundle: /root/certs/bundle
+```
+
+The auth option is **optional** as there are use cases (i.e. a mirror that only permits pulls) for which authentication may not be desired. There are currently 2 possible auth providers, "silly" and "token", only one auth provider may be configured at the moment:
+
+### silly
+
+The "silly" auth is only for development purposes. It simply checks for the existence of the "Authorization" header in the HTTP request, with no regard for the value of the header. If the header does not exist, it will respond with a challenge response, echoing back the realm, service, and scope that access was denied for. 
+
+The values of the ```realm``` and ```service``` options do not matter, however both options are **required**
```

Comment:
Just say that they set the realm and service for the authentication response. Sure, they don't matter, but that's a judgement call. :)